### PR TITLE
docs: leverage module spec

### LINKED ISSUE #95: x/leverage: Documentation and Specification
## Summary

Ensure `x/leverage` has specifications defined and included in the module root along with any relevant READMEs.

----

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by toteki**
## Description

Leverage module spec

closes #95 

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/spec/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - [Reserves](01_concepts.md#Reserves)
+    - [Liquidation](01_concepts.md#Liquidation)
+    - Important Derived Values:
+        - [uToken Exchange Rate](01_concepts.md#uToken-Exchange-Rate)
+        - [Borrow Utilization](01_concepts.md#Borrow-Utilization)
+        - [Borrow Limit](01_concepts.md#Borrow-Limit)
+        - [Borrow APY](01_concepts.md#Borrow-APY)
+        - [Lending APY](01_concepts.md#Lending-APY)
+        - [Close Factor](01_concepts.md#Close-Factor)
+        - [Market Size](01_concepts.md#Market-Size)
```
Comment (toteki) [inline x/leverage/spec/README.md:26]: I would like to ask in advance whether these sections belong in the spec or are safe being only in the ADRs.

They're all written but can be trimmed down or removed if not necessary.

**Review by adamewozniak — COMMENTED** [review]
Just a couple nitpicks 👍

File: x/leverage/spec/README.md (hunk 1 above)
Comment (alexanderbez) [reply x/leverage/spec/README.md:26]: @toteki not totally sure I follow the question?

File: x/leverage/spec/01_concepts.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# Overview
+
+This document covers basic concepts and math that determine the `leverage` module's behavior.
+
+## Accepted Assets
+
+At the foundation of the `leverage` module is the [Token Registry](02_state.md#Token-Registry), which contains a list of accepts base asset types.
```
Comment (alexanderbez) [inline x/leverage/spec/01_concepts.md:7]:
```suggestion
At the foundation of the `leverage` module is the [Token Registry](02_state.md#Token-Registry), which contains a list of accepted types.
```

**Review by alexanderbez — APPROVED** [review]
Excellent work @toteki 👏

File: x/leverage/spec/README.md (hunk 1 above)
Comment (toteki) [reply x/leverage/spec/README.md:26]: Was mainly wondering if I went into more detail than necessary at the end of `01_concepts.md`. If it doesn't look like a problem then nothing needs to be done there.

**Pull request merged**